### DISCUSSION TITLE
Can not access to vuex from external js file.

### DISCUSSION THREAD

**Discussion by leostereo** [quasarframework/quasar#9467] [General - Components / Directives / etc]
Sorry guys , I know there are a lot of similar post but anyone worked for me:
I can properly acces to vuex from my vue file component.
I created a function that stores api calls result into my vuex store.
Now I would like to move this function to  an external file but can not access to my vuex.
Already tried importing the module into my js file , this is what I have:
my index.js file inside store folder:

```
import Vue from 'vue'
import Vuex from 'vuex'

import julepx from "./julepx";
Vue.use(Vuex)

export default function (/* { ssrContext } */) {
  const Store = new Vuex.Store({
    modules: {
       julepx
    },

    // enable strict mode (adds overhead!)
    // for dev mode only
    strict: process.env.DEBUGGING
  })

  return Store
}
```
and this is my external js file inside services folder.
```
import { api } from "boot/axios";
import Store from "../store";

function getUserStatus() {
  api
    .get("/dashboard/user/100")
    .then(response => {
      console.log(JSON.stringify(response.data));
      Store.commit("julepx/UPDATE_USER", response.data);
    })
    .catch(() => {
        console.log("errror");
    });
}

export default getUserStatus;
```

Thanks ... you always help me ...

**Comment by metalsadman** [comment]
https://forum.quasar-framework.org/topic/6498/access-vuex-inside-javascript

**Comment by leostereo** [comment]
thanks @metalsadman.
I made the changes sugested on link , but it still not working.
This is last version:

index.js (store.index.js)
```
import Vue from 'vue'
import Vuex from 'vuex'

// import example from './module-example'
import julepx from "./julepx";


Vue.use(Vuex)

let Store = null;
export default function (/* { ssrContext } */) {
  const Store = new Vuex.Store({
    modules: {
       julepx
    },

    // enable strict mode (adds overhead!)
    // for dev mode only
    strict: process.env.DEBUGGING
  })

  return Store
}
export { Store };
```

service.js
```
import { api } from "boot/axios";
import { Store } from "../store";

function getUserStatus() {
  api
    .get("/dashboard/user/100")
    .then(response => {
      console.log(JSON.stringify(response.data));
      Store.commit("julepx/UPDATE_USER", response.data);
      //also tryed this.$store ....
    })
    .catch((err) => {
        console.log(err);
    });
}
export default getUserStatus;
```
im getting:
`TypeError: _store__WEBPACK_IMPORTED_MODULE_1__.Store is null,`
btw : Do I need to import also the module and subfolder modules ? 
Just in case I will try with an old code from a previous project with raw vuejs. (store single file) ,and share my results.
Thanks
Leo.

**Reply by hawkeye64** [reply]
@leostereo remove the `const` from this line so it's assigned to the global being exported
```js
const Store = new Vuex.Store({
```
to
```js
Store = new Vuex.Store({
```

**Reply by leostereo** [reply]
yes ... that fixed it ,
Thanks!!
